### PR TITLE
chore: code coverage and minor cleanup

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

Additional code coverage for #279 and minor cleanup including:
- Types pkg tests
- Make InterchainAccountPretty private/unexposed
- Removed `types/querier.go` (unused) 
- Enabled previously commented out test: InitInterchainAccount (IsBound)

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/apps/27-interchain-accounts/keeper/account.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ func (k Keeper) InitInterchainAccount(ctx sdk.Context, connectionID, counterpart
 		return err
 	}
 
-	if k.IsBound(ctx, portID) {
+	if k.portKeeper.IsBound(ctx, portID) {
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/keeper/account.go:25]: Based off of #288, is this the intended usage? @seantking @colin-axner 

I've added `IsBound` to expected keeper interface and re-able the previously commented out test.

File: modules/apps/27-interchain-accounts/types/account_test.go [hunk 2]
```diff
@@ -172,20 +178,14 @@ func (suite *TypesTestSuite) TestGenesisAccountValidate() {
...
-		Sequence:      interchainAcc.Sequence,
-		AccountOwner:  interchainAcc.AccountOwner,
-	})
+	interchainAcc := types.NewInterchainAccount(baseAcc, suite.chainB.SenderAccount.GetAddress().String())
+	bz, err := interchainAcc.MarshalYAML()
 	suite.Require().NoError(err)
 
-	bz1, err := interchainAcc.MarshalYAML()
-	suite.Require().Equal(string(bz), string(bz1))
+	expected := fmt.Sprintf("address: %s\npublic_key: \"\"\naccount_number: 0\nsequence: 0\naccount_owner: %s\n", suite.chainA.SenderAccount.GetAddress(), suite.chainB.SenderAccount.GetAddress())
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/types/account_test.go:187]: Making `interchainAccountPretty` private resulted in having to re-write this test.
This is adapted from: https://github.com/cosmos/cosmos-sdk/blob/master/x/auth/types/account_test.go#L123

Comment (seantking) [reply modules/apps/27-interchain-accounts/types/account_test.go:187]: yeah that's why I made it public :D Is it not better to leave it the other way?

**Review by seantking — APPROVED** [review]

File: modules/apps/27-interchain-accounts/types/account_test.go (hunk 2 above)
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/types/account_test.go:187]: I think it makes more sense to maintain a "prettifying" struct as private, an external party should be unaware of its existence. 

Probably best to opt for consistency with the sdk.

File: modules/apps/27-interchain-accounts/keeper/account.go (hunk 1 above)
Comment (colin-axner) [reply modules/apps/27-interchain-accounts/keeper/account.go:25]: Yes it is :+1: Thanks!

File: modules/apps/27-interchain-accounts/types/account_test.go (hunk 2 above)
Comment (colin-axner) [reply modules/apps/27-interchain-accounts/types/account_test.go:187]: consistency with sdk is my preference

**Review by colin-axner — APPROVED** [review]

**Pull request merged**